src/data/list/cycle.lean

```diff
@@ -612,6 +618,21 @@ fintype.subtype (((finset.univ : finset {s : cycle α // s.nodup}).map
   (function.embedding.subtype _)).filter cycle.nontrivial)
   (by simp)
 
+/--
+The `finset` of lists that can make the cycle.
+-/
+def lists (s : cycle α) : finset (list α) :=
+quotient.lift_on' s (λ l, (l.permutations.filter (λ (l' : list α), (l' : cycle α) = s)).to_finset) $
```

Comment:
Can you add a lemma that states what's special about this definition? Something like `mem_lists_iff_coe_eq`